crates/bevy_ui/src/ui_node.rs

```diff
@@ -1498,6 +1497,8 @@ impl From<Color> for BackgroundColor {
 #[derive(Component, Clone, Debug, Reflect, Default)]
 #[reflect(Component, Default)]
 pub struct UiTextureAtlasImage {
+    /// The tint color used to draw the atlas image
+    pub color: Color,
```

Comment:
The color should stored in a separate component I think, shared by all the foreground extraction functions, called something like `UiColor`, `ColorTint`, `ImageColor` or `ForegroundColor`.

Reply:
I think I favour `UiColor` or `ForegroundColor`, as `ColorTint` seems a bit obscure and `ImageColor` is a bit too specific when we might have some non-image foreground object we want to draw.

Reply:
I think a separate component instead of a `color` field is the wrong approach, because a single entity may have multiple components that each need their own color, or some components may need different color behavior (for example, the text component has one color per `TextSection`).